### PR TITLE
SDI-477: ✨ Switch for listener to convert to our objects instead

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt [hunk 1]
```diff
@@ -7,21 +7,19 @@ import org.springframework.stereotype.Service
 @Service
 class InboundMessageListener(private val inboundMessageService: InboundMessageService, private val objectMapper: ObjectMapper) {
 
-  @SqsListener(queueNames = ["inboundqueue"], factory = "hmppsQueueContainerFactoryProxy")
-  fun processMessage(rawMessage: String?) {
-    val (Message) = objectMapper.readValue(rawMessage, Message::class.java)
-    val event = objectMapper.readValue(Message, HmppsEvent::class.java)
+  @SqsListener("inboundqueue", factory = "hmppsQueueContainerFactoryProxy")
```
Comment (petergphillips) [inline test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt:10]: looks a bit neater without `queueNames` and an array

File: test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt [hunk 2]
```diff
@@ -7,21 +7,19 @@ import org.springframework.stereotype.Service
...
+  fun processMessage(message: Message) {
```
Comment (petergphillips) [inline test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt:11]: can convert directly into our `Message`

File: test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt [hunk 3]
```diff
@@ -7,21 +7,19 @@ import org.springframework.stereotype.Service
...
+    val event = objectMapper.readValue(message.Message, HmppsEvent::class.java)
```
Comment (petergphillips) [inline test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt:12]: we _could_ add a custom converter to convert into a specific type as well

Comment (mikehalmamoj) [reply test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt:12]: Sounds like a good idea to me.

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**